modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -0,0 +1,1229 @@
+                                 (azimuthal_tangent.size() == 0 ? ((Real)j / (Real)angle_number)
+                                                                : (azimuthal_tangent[j] / 2.0));
+        // cell_boundary_p(s) are the points on the cell's six boundries (flat sides) at
+        // differnt azimuthal angles
+        const Real pin_azimuthal_p_x =
+            cell_boundary_p_x * bin_radial_distance /
+            std::sqrt(Utility::pow<2>(cell_boundary_p_x) + Utility::pow<2>(cell_boundary_p_y));
+        const Real pin_azimuthal_p_y =
+            cell_boundary_p_y * bin_radial_distance /
+            std::sqrt(Utility::pow<2>(cell_boundary_p_x) + Utility::pow<2>(cell_boundary_p_y));
+
+        // pin_azimuthal_p are the points on the bins towards differnt azimuthal angles, at
+        // differnt intervals; excluding the ones produced by pin_corner_p
+        Point p = Point(pin_azimuthal_p_x, pin_azimuthal_p_y, 0.0);
+        (*nodes)[*node_id] = mesh->add_point(p, *node_id);
+      }
+    }
+  }
+  return dynamic_pointer_cast<ReplicatedMesh>(mesh);
+}
+
+// 'background_nodes' creates nodes for the ring-to-polygon transition region (i.e., background) of
+// a single slice.
+std::unique_ptr<ReplicatedMesh>
+PolygonMeshGeneratorBase::background_nodes(std::unique_ptr<ReplicatedMesh> mesh,
+                                           const unsigned int num_sectors_per_side,
+                                           const unsigned int background_intervals,
+                                           const Real background_corner_distance,
+                                           const Real background_corner_radial_interval_length,
+                                           dof_id_type * node_id,
+                                           const Real corner_p[2][2],
+                                           const Real corner_to_corner,
+                                           std::vector<Node *> * nodes,
+                                           const Real background_in,
+                                           const std::vector<Real> azimuthal_tangent)
+{
+  unsigned int angle_number =
+      azimuthal_tangent.size() == 0 ? num_sectors_per_side : (azimuthal_tangent.size() - 1);
+  for (unsigned int k = 0; k < (background_intervals); k++)
+  {
+    (*node_id)++;
+    const Real background_corner_p_x =
+        background_corner_distance / (0.5 * corner_to_corner) * corner_p[0][0] *
+        (background_in + (k + 1) * background_corner_radial_interval_length) /
+        background_corner_distance;
+    const Real background_corner_p_y =
+        background_corner_distance / (0.5 * corner_to_corner) * corner_p[0][1] *
+        (background_in + (k + 1) * background_corner_radial_interval_length) /
+        background_corner_distance;
+
+    // background_corner_p(s) are the points in the background region, on the bins towards the six
+    // corners, at differnt intervals
```

Comment:
```suggestion
    // corners, at different intervals
```
ok maybe just fix them all on your side please